tools/linker/CoreMarkStep.cs

```diff
@@ -281,36 +281,23 @@ protected override MethodDefinition MarkMethod (MethodReference reference)
 			return method;
 		}
 
-		protected override void MarkInterfaceImplementation (TypeDefinition type, InterfaceImplementation iface)
+		protected override bool ShouldMarkInterfaceImplementation (TypeDefinition type, InterfaceImplementation iface, TypeDefinition resolvedInterfaceType)
 		{
 			if (RegisterProtocols) {
 				// If we're registering protocols, we can remove interfaces that represent protocols.
 				// The linker will automatically mark interfaces a class implements, but we have to
 				// override the linker behavior for interfaces that represent protocols for those 
 				// interfaces to be removed.
-				var mark = false;
-				var interfaceType = iface.InterfaceType.Resolve ();
 
-				var isProtocol = type.IsNSObject (LinkContext) && interfaceType.HasCustomAttribute (LinkContext, Namespaces.Foundation, "ProtocolAttribute");
+				var isProtocol = type.IsNSObject (LinkContext) && resolvedInterfaceType.HasCustomAttribute (LinkContext, Namespaces.Foundation, "ProtocolAttribute");
 
-				if (IgnoreScope (type.Scope)) {
-					// We're not linking the current assembly, which means the interface should be marked.
-					mark = true;
-				} else if (!isProtocol) {
-					// We only skip interfaces that represent protocols.
-					mark = true;
-				}
-
-				if (!mark) {
-					if (isProtocol)
-						LinkContext.StoreProtocolMethods (interfaceType);
```

Comment:
You're removing the call to `LinkContext.StoreProtocolMethods`; this is required, because the static registrar needs access to these methods even if they've been linked away.

Reply:
It's still there a few lines below, unless I screwed up simplifying the if logic: `StoreProtocolMethods()` was only called if `isProtocol == true` and `mark == false`, and the latter only happens when `IgnoreScope () == true` or `isProtocol == false` so should be fine the way I have it :)

Reply:
Ah, nvm, I read the diff completely wrong.